src/libs/actions/Session/index.js

```diff
@@ -72,24 +74,33 @@ function createAccount(login) {
  */
 function signOut() {
     Log.info('Flushing logs before signing out', true, {}, true);
+
     if (credentials && credentials.autoGeneratedLogin) {
-        // Clean up the login that we created
-        DeprecatedAPI.DeleteLogin({
+        const optimisticData = [
+            {
+                onyxMethod: 'set',
+                key: ONYXKEYS.SESSION,
+                value: null,
+            },
+            {
+                onyxMethod: 'set',
+                key: ONYXKEYS.CREDENTIALS,
+                value: null,
+            },
+        ];
+        API.write('LogOut', {
+            // Send current authToken because we will immediately clear it once triggering this command
+            authToken: NetworkStore.getAuthToken(),
             partnerUserID: credentials.autoGeneratedLogin,
             partnerName: CONFIG.EXPENSIFY.PARTNER_NAME,
             partnerPassword: CONFIG.EXPENSIFY.PARTNER_PASSWORD,
             shouldRetry: false,
-        })
-            .then((response) => {
-                if (response.jsonCode === CONST.JSON_CODE.SUCCESS) {
-                    return;
-                }
-
-                Onyx.merge(ONYXKEYS.SESSION, {error: response.message});
-            });
+        }, {optimisticData});
+    } else {
+        Onyx.set(ONYXKEYS.SESSION, null);
+        Onyx.set(ONYXKEYS.CREDENTIALS, null);
```

Comment:
This is the same as the `optimisticData` and we could probably DRY this up somehow. I don't think it would be too terrible to call `LogOut` without an `autoGeneratedLogin` and have it always return `jsonCode: 200`? It would be a useless API call, but would clean up the logic here (which seems worth it).

Reply:
Remove the requirement to pass `autoGeneratedLogin` and `LogOut` would be an API that does "any tasks to perform when a user logs out". Even if that API command just logs that a user is logging out it seems OK (probably more things will be added in the future that the user needs to do when they log out).

Reply:
Cool making `autoGeneratedLogin` optional & consolidating 👍